src/main/java/org/broadinstitute/consent/http/resources/DataAccessRequestResource.java

```diff
@@ -126,6 +130,27 @@ public Response createDataAccessRequest(
     }
   }
 
+  @VisibleForTesting
+  public void validateInternalCollaborators(DataAccessRequest payload, User requestingUser) {
+    Integer institution = requestingUser.getInstitutionId();
+    List<Collaborator> internalCollaborators = payload.getData().getInternalCollaborators();
+    if (internalCollaborators != null && !internalCollaborators.isEmpty()) {
+      for (Collaborator collaborator : internalCollaborators) {
+        // throws not found exception
+        User collabUser = findUserByEmail(collaborator.getEmail());
+        if (!Objects.equals(collabUser.getInstitutionId(), institution)) {
+          throw new BadRequestException(
+              "Collaborator " + collaborator.getEmail() + " is not part of the same institution");
+        }
+        List<LibraryCard> libraryCards = collabUser.getLibraryCards();
+        if (libraryCards == null || libraryCards.isEmpty()) {
```

Comment:
I think `libraryCards` can't be `null` here.
```suggestion
        if (libraryCards.isEmpty()) {
```

Reply:
Why not? I see nothing on the User guaranteeing it can't be null

Reply:
The code is recent so maybe your branch is behind, the current code in `User` looks like
```
  public List<LibraryCard> getLibraryCards() {
    if (this.libraryCards == null) {
      this.libraryCards = new ArrayList<>();
    }
    return this.libraryCards;
  }
```